The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4274207921/jobs/7440654584) [(plain)](https://github.com/rust-lang/rust/commit/6a1920b1954629b97a5f47af9d3289b3aa1879c2/checks/11604302527/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking clippy_lints v0.1.69 (/checkout/src/tools/clippy/clippy_lints)
error[E0425]: cannot find value `from_fn` in module `sym`
  --> src/tools/clippy/clippy_lints/src/operators/cmp_owned.rs:55:58
   |
55 |                 } else if cx.tcx.is_diagnostic_item(sym::from_fn, did) {
   |                                                          ^^^^^^^ not found in `sym`
help: consider importing one of these items
   |
1  | use core::array::from_fn;
   |
---
   |
     and 1 other candidate
help: if you import `from_fn`, refer to it directly
   |
55 -                 } else if cx.tcx.is_diagnostic_item(sym::from_fn, did) {
55 +                 } else if cx.tcx.is_diagnostic_item(from_fn, did) {

error[E0425]: cannot find value `from_fn` in module `sym`
  --> src/tools/clippy/clippy_lints/src/unnecessary_owned_empty_strings.rs:58:59
   |
   |
58 |                         if cx.tcx.is_diagnostic_item(sym::from_fn, fun_def_id);
   |                                                           ^^^^^^^ not found in `sym`
help: consider importing one of these items
   |
1  | use core::array::from_fn;
   |
---
   |
     and 1 other candidate
help: if you import `from_fn`, refer to it directly
   |
58 -                         if cx.tcx.is_diagnostic_item(sym::from_fn, fun_def_id);
58 +                         if cx.tcx.is_diagnostic_item(from_fn, fun_def_id);

error[E0425]: cannot find value `from_fn` in module `sym`
   --> src/tools/clippy/clippy_lints/src/useless_conversion.rs:164:63
    |
    |
164 | ...                   if cx.tcx.is_diagnostic_item(sym::from_fn, def_id);
    |                                                         ^^^^^^^ not found in `sym`
help: consider importing one of these items
    |
1   | use core::array::from_fn;
    |
---
    |
      and 1 other candidate
help: if you import `from_fn`, refer to it directly
    |
164 -                             if cx.tcx.is_diagnostic_item(sym::from_fn, def_id);
164 +                             if cx.tcx.is_diagnostic_item(from_fn, def_id);

    Checking rustc-workspace-hack v1.0.0 (/checkout/src/tools/rustc-workspace-hack)
For more information about this error, try `rustc --explain E0425`.
error: could not compile `clippy_lints` due to 3 previous errors
```

</details>

--- a/compiler/rustc_hir/src/lang_items.rs
+++ b/compiler/rustc_hir/src/lang_items.rs
@@ -302,8 +302,6 @@ language_item_table! {
     Context,                 sym::Context,             context,                    Target::Struct,         GenericRequirement::None;
     FuturePoll,              sym::poll,                future_poll_fn,             Target::Method(MethodKind::Trait { body: false }), GenericRequirement::None;
 
-    FromFrom,                sym::from,                from_fn,                    Target::Method(MethodKind::Trait { body: false }), GenericRequirement::None;
-
     OptionSome,              sym::Some,                option_some_variant,        Target::Variant,        GenericRequirement::None;
     OptionNone,              sym::None,                option_none_variant,        Target::Variant,        GenericRequirement::None;
 

--- a/compiler/rustc_span/src/symbol.rs
+++ b/compiler/rustc_span/src/symbol.rs
@@ -740,6 +740,7 @@ symbols! {
         frem_fast,
         from,
         from_desugaring,
+        from_fn,
         from_iter,
         from_method,
         from_output,

--- a/library/core/src/convert/mod.rs
+++ b/library/core/src/convert/mod.rs
@@ -542,7 +542,7 @@ pub trait Into<T>: Sized {
 #[const_trait]
 pub trait From<T>: Sized {
     /// Converts to this type from the input type.
-    #[lang = "from"]
+    #[rustc_diagnostic_item = "from_fn"]
     #[must_use]
     #[stable(feature = "rust1", since = "1.0.0")]
     fn from(value: T) -> Self;

--- a/src/tools/clippy/clippy_lints/src/operators/cmp_owned.rs
+++ b/src/tools/clippy/clippy_lints/src/operators/cmp_owned.rs
@@ -49,10 +49,10 @@ fn check_op(cx: &LateContext<'_>, expr: &Expr<'_>, other: &Expr<'_>, left: bool)
             (arg, arg.span)
         },
         ExprKind::Call(path, [arg])
-            if path_def_id(cx, path).map_or(false, |id| {
-                if match_def_path(cx, id, &paths::FROM_STR_METHOD) {
+            if path_def_id(cx, path).map_or(false, |did| {
+                if match_def_path(cx, did, &paths::FROM_STR_METHOD) {
                     true
-                } else if cx.tcx.lang_items().from_fn() == Some(id) {
+                } else if cx.tcx.is_diagnostic_item(sym::from_fn, did) {
                     !is_copy(cx, typeck.expr_ty(expr))
                 } else {
                     false

--- a/src/tools/clippy/clippy_lints/src/unnecessary_owned_empty_strings.rs
+++ b/src/tools/clippy/clippy_lints/src/unnecessary_owned_empty_strings.rs
@@ -7,6 +7,7 @@ use rustc_hir::{BorrowKind, Expr, ExprKind, LangItem, Mutability};
 use rustc_lint::{LateContext, LateLintPass};
 use rustc_middle::ty;
 use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::sym;
 
 declare_clippy_lint! {
     /// ### What it does
@@ -54,7 +55,7 @@ impl<'tcx> LateLintPass<'tcx> for UnnecessaryOwnedEmptyStrings {
                         );
                 } else {
                     if_chain! {
-                        if Some(fun_def_id) == cx.tcx.lang_items().from_fn();
+                        if cx.tcx.is_diagnostic_item(sym::from_fn, fun_def_id);
                         if let [.., last_arg] = args;
                         if let ExprKind::Lit(spanned) = &last_arg.kind;
                         if let LitKind::Str(symbol, _) = spanned.node;

--- a/src/tools/clippy/clippy_lints/src/useless_conversion.rs
+++ b/src/tools/clippy/clippy_lints/src/useless_conversion.rs
@@ -161,7 +161,7 @@ impl<'tcx> LateLintPass<'tcx> for UselessConversion {
                         }
 
                         if_chain! {
-                            if Some(def_id) == cx.tcx.lang_items().from_fn();
+                            if cx.tcx.is_diagnostic_item(sym::from_fn, def_id);
                             if same_type_and_consts(a, b);
 
                             then {

